mllib/src/main/scala/org/apache/spark/mllib/util/MLUtils.scala

```diff
@@ -82,6 +82,8 @@ object MLUtils {
           val value = indexAndValue(1).toDouble
           (index, value)
         }.unzip
+        require(indices.size == 0 || indices(0) >= 0,
```

Comment:
Ah right, missed that in the fold. Would it make more sense to check the value where it's read rather than after the -1?